### ISSUE TITLE
[CRASH] update_rtpengines (force_test=0) at /build/opensips-3.4.4~20240229~e50c6736e/modules/rtpengine/rtpengine.c:1843

### ISSUE THREAD

**Issue by digipigeon** [OpenSIPS/opensips#3323]
**OpenSIPS version you are running**

```
version: opensips 3.4.4 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, Q_MALLOC, F_MALLOC, HP_MALLOC, DBG_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: e50c6736e
main.c compiled on  with gcc 11
```

**Crash Core Dump**

[core-dump-bt-full.txt](https://github.com/OpenSIPS/opensips/files/14469732/core-dump-bt-full.txt)

**Describe the traffic that generated the bug**

**To Reproduce**

There are perodic `update_rtpengines` calls, based on the crash data, I assume it was related to this incident, however this was from a production server, and we do not have any specific steps to replicate the issue.

**Relevant System Logs**

I am sorry we do not have any related logs to this event.

**OS/environment information**
 - Operating System:  Ubuntu 22.04
 - OpenSIPS installation:  Nightly Package
 - other relevant information:

**Additional context**

**Comment by digipigeon** [comment]
2 more examples from the same server with slightly different bt paths.

[core-dump-bt-full2.txt](https://github.com/OpenSIPS/opensips/files/14469764/core-dump-bt-full2.txt)
[core-dump-bt-full3.txt](https://github.com/OpenSIPS/opensips/files/14469765/core-dump-bt-full3.txt)

**Comment by digipigeon** [comment]
This has been fixed in 831cdd5
